Source/Core/AudioCommon/XAudio2Stream.cpp

```diff
@@ -58,21 +72,38 @@ void StreamingVoiceContext::SubmitBuffer(PBYTE buf_data)
 
 StreamingVoiceContext::StreamingVoiceContext(IXAudio2* pXAudio2, Mixer* pMixer,
                                              Common::Event& pSyncEvent)
-    : m_mixer(pMixer), m_sound_sync_event(pSyncEvent),
-      xaudio_buffer(new BYTE[NUM_BUFFERS * BUFFER_SIZE_BYTES]())
+    : m_mixer(pMixer), m_sound_sync_event(pSyncEvent)
 {
   WAVEFORMATEXTENSIBLE wfx = {};
-
   wfx.Format.wFormatTag = WAVE_FORMAT_EXTENSIBLE;
   wfx.Format.nSamplesPerSec = m_mixer->GetSampleRate();
-  wfx.Format.nChannels = 2;
-  wfx.Format.wBitsPerSample = 16;
+
+  if (m_use_surround)
+  {
+    xaudio_buffer = std::unique_ptr<BYTE[]>(new BYTE[NUM_BUFFERS * BUFFER_SIZE_BYTES_SURROUND]);
+
+    wfx.Format.nChannels = 6;
+    wfx.Format.wBitsPerSample = 32;
+    wfx.Format.cbSize = 22;
```

Comment:
This number is rather magical, can you show how you derived it?

Reply:
Sure. I got them from here:
https://msdn.microsoft.com/en-us/library/windows/desktop/dd390971(v=vs.85).aspx

Maybe just linking to this page is enough?

Reply:
I meant that you should calculate it using something similar to this line: `wfx.Format.cbSize = sizeof(WAVEFORMATEXTENSIBLE) - sizeof(WAVEFORMATEX);`

Reply:
```cpp
sizeof(wfx.Samples) + sizeof(wfx.dwChannelFormat) + sizeof(wfx.SubFormat);
//        2                          4                          16
```
would be the equivalent.

Reply:
And unless Windows does something less than great with the packing of that struct (I haven't checked in the header), the padding will occur between `Samples` and `dwChannelMask` if I am not mistaken in my alignment math.